### DISCUSSION TITLE
s3 sdk GetObject: is there a way to gzip decompress the content automatically

### DISCUSSION THREAD

**Discussion by Chandrian** [awslabs/aws-sdk-rust#451] [Q&A] [answered]
Hello,

Wondering if there's a way to decompress the content using GetObject if the content-encoding is set to GZIP?
Basically, thinking of how [reqwest](https://docs.rs/reqwest/0.11.9/reqwest/struct.ClientBuilder.html#method.gzip) works.

Right now I'm using flate2, but having issues as it errors out with a corrupt data error, but matching on it I actually get the right output.. not clean so looking for something better.

Thanks for any insight.

**Comment by rcoh** [comment, marked as answer]
I don't believe this is something that we've tried—there's no good pattern to do this right now, but we'll get this on the backlog
